File: smart-home-backend/src/main/java/com/bsep/smart/home/model/Property.java
package com.bsep.smart.home.model;

import lombok.*;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;
import java.util.List;

@Getter
@Setter
@FieldDefaults(level = AccessLevel.PRIVATE)
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "property")
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
@Entity
@SuperBuilder
public class Property extends BaseEntity {
    String name;
    String address;
    @ManyToOne
    Person owner;
    @ManyToMany
    List<Person> members;
    @OneToMany(mappedBy = "property", cascade = CascadeType.ALL)
    List<Device> devices;

}

File: Smart-home-frontend/src/app/shared/model/property.ts
import { User } from './user.model';
import {Device} from "./device.model";

export class Property {
  constructor(
    public uuid: string,
    public name: string,
    public owner: User,
    public address: string,
    public members: User[],
    public devices: Device[]
  ) {}
}
